macros/edr/materializations/tests/test.sql

```diff
@@ -0,0 +1,242 @@
+    {% set sample_row = anomaly_scores_rows[0] %}
+    {% do elementary.cache_elementary_test_results_rows([elementary.get_anomaly_test_result_row(flattened_test, sample_row, anomaly_scores_rows)]) %}
+  {% else %}
+    {% set test_result_metrics_rows = {} %}
+    {% for anomaly_score_row in anomaly_scores_rows %}
+      {% do test_result_metrics_rows.setdefault(anomaly_score_row.metric_name, []) %}
+      {% do test_result_metrics_rows[anomaly_score_row.metric_name].append(anomaly_score_row) %}
+    {% endfor %}
+
+    {% set elementary_test_results_rows = [] %}
+    {% for metric_rows in test_result_metrics_rows.values() %}
+      {% for metric_row in metric_rows %}
+        {% do elementary_test_results_rows.append(elementary.get_anomaly_test_result_row(flattened_test, metric_row, metric_rows)) %}
+      {% endfor %}
+    {% endfor %}
+    {% do elementary.cache_elementary_test_results_rows(elementary_test_results_rows) %}
+  {% endif %}
+
+  {% do context.update({"sql": elementary.get_anomaly_query(flattened_test)}) %}
+{% endmacro %}
+
+{% macro handle_schema_changes_test(flattened_test) %}
+  {% set schema_snapshots_tables_cache = elementary.get_cache("tables").get("schema_snapshots") %}
+  {% set schema_snapshots_table = elementary.get_elementary_test_table(flattened_test.name, 'schema_changes') %}
+  {% if schema_snapshots_table %}
+    {% do schema_snapshots_tables_cache.append(schema_snapshots_table) %}
+  {% endif %}
+
+  {% set elementary_test_results_rows = [] %}
+  {% set schema_changes_rows = elementary.query_test_result_rows() %}
+  {% for schema_changes_row in schema_changes_rows %}
+    {% do elementary_test_results_rows.append(elementary.get_schema_changes_test_result_row(flattened_test, schema_changes_row)) %}
+  {% endfor %}
+  {% do elementary.cache_elementary_test_results_rows(elementary_test_results_rows) %}
+{% endmacro %}
+
+{% macro handle_dbt_test(flattened_test) %}
+  {% set result_rows = elementary.query_test_result_rows(sample_limit=elementary.get_config_var('test_sample_row_count')) %}
+  {% set elementary_test_results_row = elementary.get_dbt_test_result_row(flattened_test, result_rows) %}
+  {% do elementary.cache_elementary_test_results_rows([elementary_test_results_row]) %}
+{% endmacro %}
+
+{% macro get_elementary_test_type(flattened_test) %}
+  {% if flattened_test.test_namespace == "elementary" %}
+    {% if flattened_test.short_name.endswith("anomalies") %}
+      {% do return("anomaly_detection") %}
+    {% elif flattened_test.short_name == 'schema_changes' %}
+      {% do return("schema_change") %}
+    {% endif %}
+  {% endif %}
+  {% do return("dbt_test") %}
+{% endmacro %}
+
+{% macro materialize_test() %}
+  {% set flattened_test = elementary.flatten_test(model) %}
+  {% set test_type = elementary.get_elementary_test_type(flattened_test) %}
+  {% set test_type_handler_map = {
+    "anomaly_detection": elementary.handle_anomaly_test,
+    "schema_change": elementary.handle_schema_changes_test
```

Comment:
I think it will better to add the dbt_test type to the type handler map as well.
You want the map to contain all the supported types there is (even if you use the dbt test type as default from the outside).